src/main/java/se/sundsvall/document/service/RegistrationNumberService.java

```diff
@@ -0,0 +1,67 @@
+package se.sundsvall.document.service;
+
+import static java.lang.String.format;
+import static java.time.OffsetDateTime.now;
+import static java.time.ZoneId.systemDefault;
+
+import java.time.OffsetDateTime;
+
+import org.springframework.stereotype.Service;
+
+import jakarta.transaction.Transactional;
+import se.sundsvall.document.integration.db.RegistrationNumberSequenceRepository;
+import se.sundsvall.document.integration.db.model.RegistrationNumberSequenceEntity;
+
+/**
+ * Class responsible for generating unique registration numbers.
+ *
+ * Registration numbers are created with the following pattern: [YYYY-MUNICIPALITY_ID-SEQUENCE]
+ *
+ * Example:
+ * If a registrationNumber is created on date 2022-10-26 for "Sundsvall municipality" (municipalityID: 2281), for the
+ * first time, the registrationNumber will be: 2022-2281-1. The next generated number will be 2022-2281-2 and so on.
+ *
+ * Every new year, the sequence will be reset to 1. (e.g. 2023-2281-1).
+ */
+@Service
+@Transactional
+public class RegistrationNumberService {
+
+	private static final int SEQUENCE_START = 1;
+	private static final String REGISTRATION_NUMBER_TEMPLATE = "%s-%s-%s"; // [YYYY-MUNICIPALITY_ID-SEQUENCE]
+
+	private final RegistrationNumberSequenceRepository registrationNumberSequenceRepository;
+
+	public RegistrationNumberService(RegistrationNumberSequenceRepository registrationNumberSequenceRepository) {
+		this.registrationNumberSequenceRepository = registrationNumberSequenceRepository;
+	}
+
+	public String generateRegistrationNumber(String municipalityId) {
+
+		final var sequenceEntity = registrationNumberSequenceRepository.findByMunicipalityId(municipalityId)
+			.orElse(RegistrationNumberSequenceEntity.create()
+				.withCreated(now(systemDefault()))
```

Comment:
Yes, you are right and I was aware of that. 

I only added this row because of how getLastTouch()-method was constructed. 
But I have refactored it and removed this line now. 

In the end, the total line count was pretty much the same. But the new (refactored) way is probably better and more readable.